pkg/registration/spoke/options.go

```diff
@@ -74,6 +76,10 @@ func (o *SpokeAgentOptions) AddFlags(fs *pflag.FlagSet) {
 			"the value of --cluster-signing-duration command-line flag of the kube-controller-manager will be used.")
 	fs.StringToStringVar(&o.ClusterAnnotations, "cluster-annotations", o.ClusterAnnotations, `the annotations with the reserve
 	 prefix "agent.open-cluster-management.io" set on ManagedCluster when creating only, other actors can update it afterwards.`)
+	fs.StringVar(&o.RegistrationAuth, "registration-auth", o.RegistrationAuth,
```

Comment:
I think we should have a drivers Options and we just need to AddFlags here?

Reply:
@qiujian16 We are currently using the `registration-auth` flag to determine the type of `registerDriver` to set up. See lines 199-202 in pkg/registration/spoke/spokeagent.go. 

Are you suggesting that we should add another flag to capture the type of the registration driver as well?

Reply:
the flag name is fine. I think you could put all these flags adding func to the register directory with a new Option struct. And the code here just to call driverOptions.AddFlags(fs)